protocol/signalfx/additionalspantags/additionalspantags.go

```diff
@@ -0,0 +1,54 @@
+package additionalspantags
+
+import (
+	"context"
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/datapoint/dpsink"
+	"github.com/signalfx/golib/event"
+	"github.com/signalfx/golib/trace"
+)
+
+type sink interface {
+	dpsink.Sink
+	trace.Sink
+}
+
+var _ sink = &AdditionalSpanTags{}
+
+// AdditionalSpanTags adds tags to every span it processes
+type AdditionalSpanTags struct {
+	tags map[string]string
+	next sink
+}
+
+// AddDatapoints is a passthrough
+func (a *AdditionalSpanTags) AddDatapoints(ctx context.Context, points []*datapoint.Datapoint) error {
+	return a.next.AddDatapoints(ctx, points)
+}
+
+// AddEvents is a passthrough
+func (a *AdditionalSpanTags) AddEvents(ctx context.Context, events []*event.Event) error {
+	return a.next.AddEvents(ctx, events)
+}
+
+// AddSpans adds tags to all of the spans that are given
+func (a *AdditionalSpanTags) AddSpans(ctx context.Context, spans []*trace.Span) error {
+	for _, s := range spans {
```

Comment:
do we want to do this only if len(a.Tags) > 0?
i guess it's not really necessary since you check that before creating this object...